### PR TITLE
Nc news 11

### PR DESCRIPTION
**Pull request by hjaox**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -97,6 +97,76 @@ describe('App Tests', () => {
...
+        describe('Added Feature: GET `/api/articles` test', () => {
+            test('topic query: filters the articles by topic value specified in the query.', () => {
+                return request(app)
+                .get('/api/articles?topic=mitch')
+                .then(({body: {articles}}) => {
+                    expect(articles.length).not.toBe(0);
+                    articles.forEach(article => {
+                        expect(article.topic).toBe('mitch');
+                    })
+                })
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:109]: This is a good test!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -97,6 +97,76 @@ describe('App Tests', () => {
...
+            })
+            test('topic query: filters the articles by topic value specified in the query and is not case-sensitive.', () => {
+                return request(app)
+                .get('/api/articles?topic=MiTcH')
+                .then(({body: {articles}}) => {
+                    expect(articles.length).not.toBe(0);
+                    articles.forEach(article => {                        
+                        expect(article.topic).toMatch(/MiTcH/i);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: This assertion is being run on the response so you could assert `expect(article.topic).toBe("mitch")` again

File: __tests__/app.test.js [hunk 3]
```diff
@@ -97,6 +97,76 @@ describe('App Tests', () => {
...
+                    })
+                })
+            })
+            test('sort_by query: sorts the articles by any valid column(defaults to created_at)', () => {
+                return request(app)
+                .get('/api/articles?sort_by=title')
+                .then(({body: {articles}})=> {
+                    expect(articles.length).not.toBe(0);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: Could be more precise here by asserting for the specific length you expect it to be, remember that the test data should always remain constant so there's no reason to expect the length to ever change

File: __tests__/app.test.js [hunk 4]
```diff
@@ -97,6 +97,76 @@ describe('App Tests', () => {
...
+                    expect(articles).toBeSortedBy('title', {descending: true});
+                    expect(articles).toEqual(articles.sort((a,b) => b.title - a.title));
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: You can remove line _127_ as it's only doing the same thing as line _126_ but long hand

File: __tests__/app.test.js [hunk 5]
```diff
@@ -97,6 +97,76 @@ describe('App Tests', () => {
...
+                    })
+                })
+            })
+            test('order query: can be set to asc for ascending', () => {
+                return request(app)
+                .get('/api/articles?sort_by=article_id&order=asc')
+                .then(({body: {articles}})=> {
+                    expect(articles.length).not.toBe(0);
+                    expect(articles).toBeSortedBy('article_id', {ascending: true});
+                    expect(articles).toEqual(articles.sort((a,b) => b.article_id - a.article_id));
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: could remove line _148_ as it's just a long hand repetition of line _147_

File: __tests__/app.test.js [hunk 6]
```diff
@@ -97,6 +97,76 @@ describe('App Tests', () => {
...
+                        expect(article.topic).toBe('mitch');
+                    })
+                })
+            })
+            test('order query: can be set to asc for ascending', () => {
+                return request(app)
+                .get('/api/articles?sort_by=article_id&order=asc')
+                .then(({body: {articles}})=> {
+                    expect(articles.length).not.toBe(0);
+                    expect(articles).toBeSortedBy('article_id', {ascending: true});
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: Looking at the docs I think you might want this to be `('article_id', { descending: false});` rather than _ascending: true_

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Good testing! Some missing tests however:
- What do you expect if the request query gives an invalid _sort_by_ ?
- What do you expect if the request query gives an invalid _order_ ?
- What do you expect if the _topic_ can't be found in the database ?
- What do you expect if the _topic_ was found in the database but has no _articles_ ?

Add these tests, they may require you to refactor your model and how you handle _sort_by_ and _order_ then submit another PR

**Review by Keb84 — APPROVED** [review]
Nice work amending your code from the feedback 🙂 merge when ready 👍

**Pull request merged**